### DISCUSSION TITLE
input file for species block

### DISCUSSION THREAD

**Discussion by RatulSabui** [epochpic/epoch#194] [Q&A]
this is the first time i'm using Data files as input for the plasma distribution. I am getting a segmentation error while trying to use it. the input deck seems to work when I hard code the density. It must  be some mistake on my part. 

Have attached the input deck and the python code i've used to generate the dat file.

Any simple matlab or python example would also do. 

Thanks a lot for the help.

[input_deck.txt](https://github.com/Warwick-Plasma/epoch/files/6320051/input_deck.txt)
[num_den_txt.txt](https://github.com/Warwick-Plasma/epoch/files/6320053/num_den_txt.txt)

**Comment by keithbennett** [comment]
Sorry for the late response. In your input deck, "las_lambda" has not been defined.

**Reply by keithbennett** [reply]
If I add this constant then I get the error:
```
 Intial condition settings mean that there are no cells where particles may
 validly be placed for species "proton". Code will now terminate.
```

**Reply by RatulSabui** [reply]
sorry for the missing constant. that was a glitch. 

The second species block was a problem all along. Could not narrow it down  (my old version of epoch could not point it out.)

Thanks for helping.

**Reply by RatulSabui** [reply]
sorry about the constant. That was just a glitch. the main mistake was in the second species block all throughout. so 2 species cannot replicate each other's density profiles while using an input data file. This is possible only when hard coding the density within the input deck.

thank you.
